creator-node/src/utils/fsUtils.ts

```diff
@@ -124,10 +119,17 @@ export async function validateStateForImageDirCIDAndReturnFileUUID(
 
   // Ensure every file exists on disk
   await Promise.all(
-    imageFiles.map(async function (imageFile: { storagePath: string }) {
-      if (!(await fs.pathExists(imageFile.storagePath))) {
+    imageFiles.map(async function (imageFile) {
+      const computedFilePath = computeFilePathInDir(
+        imageDirCID,
+        imageFile.multihash
+      )
+      if (
+        !(await fs.pathExists(computedFilePath)) &&
+        !(await fs.pathExists(imageFile.storagePath))
+      ) {
         throw new Error(
-          `No file found on disk for imageDirCID ${imageDirCID} image file at path ${imageFile.storagePath}`
+          `No file found on disk for imageDirCID ${imageDirCID} image file at computedFilePath ${computedFilePath} or fallback db storagePath ${imageFile.storagePath}`
```

Comment:
* this change is important to prevent issues when we update `type=image` rows in the Files table but not their corresponding `type=dir` rows
* we don't need to verify existence of the image dir because we already verify existence of every image file in it (file obviously can't exist if its parent dir doesn't exist)
* when verifying existence of image files, first check computed storage path and fall back to storagePath from db (which could be legacy or custom)

Reply:
great catch